pkg/client/fake/client.go

```diff
@@ -838,6 +885,81 @@ func (c *fakeClient) patch(obj client.Object, patch client.Patch, opts ...client
 	return err
 }
 
+// Applying a patch results in a deletionTimestamp that is truncated to the nearest second.
+// Check that the diff between a new and old deletion timestamp is within a reasonable threshold
+// to be considered unchanged.
+func validateTimestampDifference(newObj metav1.Object, obj metav1.Object) bool {
```

Comment:
```suggestion
func deletionTimestampEqual(newObj metav1.Object, obj metav1.Object) bool {
```
nit. I think this would read better where we use it